protocol/backend/backend.go

```diff
@@ -262,7 +262,7 @@ func (b *backend) ClearXMRDepositAddress(offerID types.Hash) {
 // HasOngoingSwapAsTaker returns nil if we have an ongoing swap with the given peer where
 // we're the xmrtaker, otherwise returns an error.
 func (b *backend) HasOngoingSwapAsTaker(remotePeer peer.ID) error {
-	swaps, err := b.swapManager.GetOngoingSwaps()
+	swaps, err := b.swapManager.GetOngoingSwapsSnapshot()
```

Comment:
This was another change that I made when I was still working on a design that kept the status channel inside the `swap.Info` structure, but it seems worth keeping. Methods that return deep copies and are therefore completely thread safe, have the word `Snapshot` on the end.